src/server.rs

```diff
@@ -113,7 +113,7 @@ fn handle_client(tcp_stream: Arc<Mutex<TcpStream>>, clients_set: &Arc<Mutex<Hash
         for (key, values) in client_deps.entries() {
             let mut dependencies_vector = Vec::new();
             for dependency in values.members() {
-                if dependency.to_string() != key.to_string() {
+                if *dependency != *key {
```

Comment:
```suggestion
                if dependency != key {
```
The compiler handles dereferencing variables for comparisons. Comparing `&T` and `T` is equivalent.